jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/resource/ResourceFactoryInternals.java

```diff
@@ -48,10 +49,17 @@
         CURRENT_WORKING_DIR = Path.of(System.getProperty("user.dir"));
 
         // The default resource factories
-        RESOURCE_FACTORIES.put("jar", new MountedPathResourceFactory());
+        MountedPathResourceFactory mountedPathResourceFactory = new MountedPathResourceFactory();
+        RESOURCE_FACTORIES.put("jar", mountedPathResourceFactory);
         PathResourceFactory pathResourceFactory = new PathResourceFactory();
         RESOURCE_FACTORIES.put("file", pathResourceFactory);
         RESOURCE_FACTORIES.put("jrt", pathResourceFactory);
+
+        URL url = ResourceFactoryInternals.class.getResource("/org/eclipse/jetty/version/build.properties");
+        if ((url != null) && !RESOURCE_FACTORIES.contains(url.getProtocol()))
+        {
+            RESOURCE_FACTORIES.put(url.getProtocol(), mountedPathResourceFactory);
```

Comment:
Why not using a new `MountedPathResourceFactory` instance?

Reply:
No real reason to, the implementation wouldn't change with each new protocol.

Reply:
An argument could be made that there should be a test here, to see if the URL points to a mounted FileSystem or not, and adding the appropriate ResourceFactory based on results.

Or using some variation of the `mountIfNeeded()` logic always here?

Reply:
If adding some logic to figure out if mounting is needed or not doesn't take too much, let's add it. Otherwise let's get these changes make it in and eventually open an enhancement bug.